databricks/koalas/tests/test_groupby.py

```diff
@@ -203,47 +203,82 @@ def test_split_apply_combine_on_series(self):
             (True, ["var", "std"]),
         ]
         funcs = [(almost, f) for almost, fs in funcs for f in fs]
-        for ddkey, pdkey in [("b", "b"), (kdf.b, pdf.b), (kdf.b + 1, pdf.b + 1)]:
+
+        for as_index in [True, False]:
+            if as_index:
+                sort = lambda df: df.sort_index()
+            else:
+                sort = lambda df: df.sort_values(list(df.columns)).reset_index(drop=True)
+
             for almost, func in funcs:
-                self.assert_eq(
-                    getattr(kdf.groupby(ddkey).a, func)().sort_index(),
-                    getattr(pdf.groupby(pdkey).a, func)().sort_index(),
-                    almost=almost,
-                )
-                self.assert_eq(
-                    getattr(kdf.groupby(ddkey), func)().sort_index(),
-                    getattr(pdf.groupby(pdkey), func)().sort_index(),
-                    almost=almost,
-                )
+                for kkey, pkey in [("b", "b"), (kdf.b, pdf.b)]:
+                    with self.subTest(as_index=as_index, func=func, key=pkey):
+                        if as_index is True or func != "std":
+                            self.assert_eq(
+                                sort(getattr(kdf.groupby(kkey, as_index=as_index).a, func)()),
+                                sort(getattr(pdf.groupby(pkey, as_index=as_index).a, func)()),
+                                almost=almost,
+                            )
+                            self.assert_eq(
+                                sort(getattr(kdf.groupby(kkey, as_index=as_index), func)()),
+                                sort(getattr(pdf.groupby(pkey, as_index=as_index), func)()),
+                                almost=almost,
+                            )
+                        else:
+                            # seems like a pandas' bug for as_index=False and func == "std"?
```

Comment:
I suspect this is a pandas' bug:

```py
>>> pdf.groupby(pdf.b, as_index=False).sum()
   b   a
0  1  13
1  2   9
2  3   8
3  4   1
4  7   6
>>> pdf.groupby(pdf.b, as_index=False).var()
   b          a
0  1   2.333333
1  2  12.500000
2  3   0.000000
3  4        NaN
4  7        NaN
>>> pdf.groupby(pdf.b, as_index=False).std()
          b         a
0  1.000000  1.527525
1  1.414214  3.535534
2  1.732051  0.000000
3  2.000000       NaN
4  2.645751       NaN
```

For only `std`, the behavior is not following the others.